nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageService.java

```diff
@@ -0,0 +1,136 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Abdoul BA<aba@nuxeo.com>
+ */
+
+package org.nuxeo.coldstorage.service;
+
+import java.time.Duration;
+
+import org.nuxeo.coldstorage.ColdStorageConstants.ColdStorageContentStatus;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.DocumentRef;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.blob.ManagedBlob;
+
+/**
+ * @since 10.10
+ */
+public interface ColdStorageService {
+
+    /**
+     * Moves the main content associated with the document of the given {@link DocumentRef} to a cold storage.
+     * <p/>
+     * The permission {@value org.nuxeo.coldstorage.ColdStorageConstants#WRITE_COLD_STORAGE} is required.
+     *
+     * @implSpec moves the main content to ColdStorage and fires an
+     *           {@value org.nuxeo.coldstorage.ColdStorageConstants#COLD_STORAGE_CONTENT_MOVED_EVENT_NAME} event.
+     * @return the updated document model if the move succeeds
+     * @throws NuxeoException if the main content is already in the cold storage, if there is no main content associated
+     *             with the given document, or if the user does not have the permissions needed to perform the action.
+     */
+    public DocumentModel moveContentToColdStorage(CoreSession session, DocumentRef documentRef);
+
+    public DocumentModel moveToColdStorage(CoreSession session, DocumentRef documentRef);
+
+    /**
+     * Requests a retrieval of the cold storage content associated with the document of the given {@link DocumentRef}.
+     *
+     * @param session the core session
+     * @param documentRef the document reference
+     * @param restoreDuration the duration that you want your cold storage content to be accessible after restoring it
+     * @apiNote This method will initiate a restoration request, calling the {@link Blob#getStream()} during this
+     *          process doesn't mean you will get the blob's content.
+     * @return the updated document model if the retrieve succeeds
+     * @throws NullPointerException if the {@code restoreDuration} parameter is {@code null}
+     * @throws NuxeoException if there is no cold storage content associated with the given document, or if it is being
+     *             retrieved
+     */
+    public DocumentModel requestRetrievalFromColdStorage(CoreSession session, DocumentRef documentRef,
+            Duration restoreDuration);
+
+    public DocumentModel retrieveFromColdStorage(CoreSession session, DocumentModel doc, Duration numberOfDays);
```

Comment:
Should `numberOfDays` be named differently?

Reply:
I propose `getRetrieveDuration` or `getDefaultRetrieveDuration`